src/lsm/manifest_log.zig

```diff
@@ -433,6 +433,7 @@ pub fn ManifestLogType(comptime Storage: type, comptime TableInfo: type) type {
                 block,
                 address,
             );
+            manifest_log.blocks.advance_head();
```

Comment:
I'm a little nervous about this since it (sort of) implies that someone else could start using this block address while the grid is still using it to write from.

Reply:
grid.write takes the block and replaces it with an uninitialized block which is safe to modify. It's the uninitialized block that I want to advance past.